ops/testing.py

```diff
@@ -1124,6 +1127,25 @@ def stop_services(
         for name in services:
             self._service_status[name] = pebble.ServiceStatus.INACTIVE
 
+    def restart_services(
+            self, services: typing.List[str], timeout: float = 30.0, delay: float = 0.1,
+    ) -> pebble.ChangeID:
+        # handle a common mistake of passing just a name rather than a list of names
+        if isinstance(services, str):
+            raise TypeError('restart_services should take a list of names, not just "{}"'.format(
+                services))
+        # TODO: handle invalid names
+        # Note: jam 2021-04-20 We don't implement ChangeID, but the default caller of this is
+        # Container.restart() which currently ignores the return value
+        known_services = self._render_services()
+        for name in services:
+            if name not in known_services:
+                # TODO: jam 2021-04-20 This needs a better error type
+                #  400 Bad Request: service "bal" does not exist
+                raise RuntimeError('400 Bad Request: service "{}" does not exist'.format(name))
+        for name in services:
```

Comment:
I don't think we expose it in testing, but I do wonder if we should have some way to find active ones, flag them as deactivated, and then iterate again and mark them as activated.